doc/docs/Python_User_Interface.md

```diff
@@ -2706,6 +2706,26 @@ slow group velocities and are absorbed poorly by [PML](Perfectly_Matched_Layer.m
 
 </div>
 
+<a id="stop_when_dft_decayed"></a>
+
+```python
+def stop_when_dft_decayed(tol=None,
+                      minimum_run_time=0,
+                      maximum_run_time=None):
+```
+
+<div class="function_docstring" markdown="1">
+
+Return a `condition` function, suitable for passing to `Simulation.run` as the `until`
+or `until_after_sources` parameter, that checks the `Simulation`'s dft objects every `dt`
+timesteps, and stops the simulation once all the field components and frequencies of *every*
+dft object have decayed by at least some tolerance `tol` (no default value). The time interval
```

Comment:
Shouldn’t we give a default tol, e.g. the tol from the tests?

Reply:
A default value would be useful but since `tol` is no longer related to the actual precision of the fields, it's not clear what a useful value should be. #1740 added a default value of `1e-11` to the `OptimizationProblem` class of the adjoint solver but that should probably also be removed and defined instead as part of the unit tests:

https://github.com/NanoComp/meep/blob/81f2081ca81d0b6f919dfc0d1c9465445a076441/python/adjoint/optimization_problem.py#L72

Reply:
Note that `stop_when_fields_decayed` also has no default value for its `decay_by` parameter.

Reply:
Wouldn't a "useful value" be one for which the tests pass? We can run more tests... but at the end of the day it's a user parameter.

Reply:
I ran some experiments with the adjoint solver on a couple of different test problems and it seems that a default value of 1e-11 is reasonable (i.e., the gradient can be computed with 5-6 digits of precision at modest grid resolutions). I made the change to the default value but It would be useful at some point to add more guidance on the appropriate choice of this parameter.